### ISSUE TITLE
Problem with localization DataGrid

### ISSUE THREAD

**Issue by kamxas** [mui-org/material-ui-x#2633]
import { createTheme, ThemeProvider } from '@material-ui/core/styles';
import { DataGrid, ruRU } from '@mui/x-data-grid';

localization working

import { createTheme, ThemeProvider } from '@mui/material/styles';
import { DataGrid, ruRU } from '@mui/x-data-grid';

localization not working

**Comment by m4theushw** [comment]
Please provide a **minimal** reproduction test case with the latest version. This would help a lot 👷 .
A live example would be perfect. [This **codesandbox.io** template](https://material-ui.com/r/issue-template) _may_ be a good starting point. Thank you!

**Comment by kamxas** [comment]
https://codesandbox.io/s/pensive-lehmann-0pw6j?file=/src/Demo.tsx  - working (only with 4.12.3 @material-ui/core with 5.0 not worked)
https://codesandbox.io/s/sweet-bhaskara-htgp6?file=/src/Demo.tsx -not working

Code is from https://mui.com/ru/components/data-grid/localization/

**Comment by m4theushw** [comment]
The v4 version of `@mui/x-data-grid` uses internally `@material-ui/core/styles`. Changing the language of the theme in `@mui/material/styles` has no effect. Changing the import below solves the problem:

```diff
-import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { createTheme, ThemeProvider } from "@material-ui/core/styles";
```

A quick note: we released v5.0.0-beta.1 with support to MUI v5 without the need to install the old libraries.
